internal/kafka/internal/services/kafkatlscertmgmt/secure_storage.go

```diff
@@ -48,18 +51,36 @@ func newSecureStorage(config *config.AWSConfig, automaticCertificateManagementCo
 		return nil, err
 	}
 	return &secureStorage{
-		secretClient: secretClient,
-		secretCache:  secretCache,
-		secretPrefix: config.SecretManager.SecretPrefix,
+		secretClient:     secretClient,
+		secretCache:      secretCache,
+		lock:             sync.Mutex{},
+		storageItemLocks: map[string]*sync.Mutex{},
+		secretPrefix:     config.SecretManager.SecretPrefix,
 	}, nil
 }
 
 func (storage *secureStorage) Lock(ctx context.Context, key string) error {
```

Comment:
We can document that Lock blocks until it can acquire the name-specific lock, and that the process of obtaining a name-specific lock is atomic, as required per the certmagic.Locker interface documentation

Reply:
done